src/System.IO.Pipelines/StreamExtensions.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    public static class StreamExtensions
+    {
+        /// <summary>
+        /// Adapts a <see cref="Stream"/> into a <see cref="IPipelineWriter"/>.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <returns></returns>
+        public static IPipelineWriter AsPipelineWriter(this Stream stream)
+        {
+            return (stream as IPipelineWriter) ?? stream.AsPipelineWriter(ArrayBufferPool.Instance);
+        }
+
+        /// <summary>
+        /// Adapts a <see cref="Stream"/> into a <see cref="IPipelineWriter"/>.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="pool"></param>
+        /// <returns></returns>
+        public static IPipelineWriter AsPipelineWriter(this Stream stream, IBufferPool pool)
+        {
+            var writer = new PipelineReaderWriter(pool);
+            writer.CopyToAsync(stream).ContinueWith((task) =>
+            {
+                if (task.IsFaulted)
+                {
+                    writer.CompleteReader(task.Exception);
+                }
+                else
+                {
+                    writer.CompleteReader();
+                }
+            });
```

Comment:
Would be better as:
```C#
writer.CopyToAsync(stream).ContinueWith((t,s) =>
{
    var innerWriter = (PipelineReaderWriter)s;
    if (t.IsFaulted) innerWriter.CompleteReader(task.Exception.InnerException);
    else innerWriter.CompleteReader();
}, writer, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```
That way, you avoid the closure/delegate allocation closing over state, the continuation is invoked as part of the antecedent task's completion (rather than queueing another work item, and the continuation will not be forced back to any custom TaskScheduler that was present.  Similar comments apply to other uses of ContinueWith in this PR.